packages/tx/README.md

```diff
@@ -17,7 +17,75 @@ Note: this `README` reflects the state of the library from `v3.0.0` onwards. See
+
+- `public static fromTxData(txData: TxData, opts: TxOptions = {})`: instantiate from a data dictionary
+- `public static fromSerializedTx(serialized: Buffer, opts: TxOptions = {})`: instantiate from a serialized tx
+- `public static fromValuesArray(values: Buffer[], opts: TxOptions = {})`: instantiate from a values array
+
+See one of the code examples on the tx types below on how to use.
+
+All types of transaction objects are frozen with `Object.freeze()` which gives you enhanced security and consistency properties when working with the instantiated object. This behavior can be modified using the `freeze` option in the constructor if needed.
+
+## Transaction Types
+
+This library supports the following transaction types ([EIP-2718](https://eips.ethereum.org/EIPS/eip-2718)):
+
+- `AccessListEIP2930Transaction` ([EIP-2930](https://eips.ethereum.org/EIPS/eip-2930), optional access lists)
+- `Transaction`, the Ethereum standard tx up to `berlin`, now referred to as legacy txs with the introduction of tx types
+
+Please note that for now you have to manually set the `hardfork` in `Common` to `berlin` to allow for typed tx instantiation, since the current `Common` release series v2 (tx type support introduced with `v2.2.0`) still defaults to `istanbul` for backwards-compatibility reasons.
+
+#### Access List Transactions (EIP-2930)
+
+- Class: `AccessListEIP2930Transaction`
+- Activation: `berlin`
+
+This is the recommended tx type starting with the activation of the `berlin` HF, see the following code snipped for an example on how to instantiate:
+
+```typescript
+import Common from '@ethereumjs/common'
+import { AccessListEIP2930Transaction } from '@ethereumjs/tx'
+
+const common = new Common({ chain: 'mainnet', hardfork: 'berlin' })
+
+const txData = {
+  "data": "0x1a8451e600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+  "gasLimit": "0x02625a00",
+  "gasPrice": "0x01",
+  "nonce": "0x00",
+  "to": "0xcccccccccccccccccccccccccccccccccccccccc",
+  "value": "0x0186a0",
+  "v": "0x01",
+  "r": "0xafb6e247b1c490e284053c87ab5f6b59e219d51f743f7a4d83e400782bc7e4b9",
+  "s": "0x479a268e0e0acd4de3f1e28e4fac2a6b32a4195e8dfa9d19147abe8807aa6f64",
+  "chainId": "0x01",
+  "accessList": [
+    {
+      "address": "0x0000000000000000000000000000000000000101",
+      "storageKeys": [
+        "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "0x00000000000000000000000000000000000000000000000000000000000060a7"
+      ]
+    }
+  ],
+  "type": "0x01"
+}
+
+const tx = AccessListEIP2930Transaction.fromTxData(txData, { common })
+```
+
+A mechanism to generate access lists from tx data based on a certain network state in not part of this library.
```

Comment:
Is there any advice that can be given about how to identify storage slots?

Reply:
@cgewecke it is possible to derive the storage keys for a contract from its own code (check out [this example](https://eth.wiki/json-rpc/API#example-14)), however getting the access list for a more complex transaction can be tricky as it can change based on the latest network state.

Reply:
I definitely think we should introduce some tools to help generate access lists though, I expect that it will be requested if it helps lower the cost of transactions. Perhaps it's better suited at the library layer in web3.js and ethers where there is a connection to an Ethereum node.

Do we think there will be JSON-RPC endpoints or helper tools from wallets like MetaMask to generate contract transactions containing access lists? Etherscan would actually be in a really good position to generate them when creating a tx on their "write contract" page.

Reply:
I think we'll relatively for sure add this as a feature to the VM - see also https://github.com/ethereumjs/ethereumjs-monorepo/issues/1152 - since this is just dropping out there on the sideline and *will* likely also have it's use cases, not sure if it is applicable for these kind of usages since there is still this "where to get the state from" problem?

Would this be another case for an RPC-based `StateManager` (this was in discussion sometime ago somewhere 😛 ) retrieving all the state by RPC on demand? This would be really cool in general, thinking about this once again (and actually might also work for use cases like MetaMask I would say for generating access lists?).